dashboard/test/models/concerns/user/email_validations_test.rb

```diff
@@ -0,0 +1,124 @@
+require 'test_helper'
+
+class EmailValidationsTest < ActiveSupport::TestCase
+  include Minitest::RSpecMocks
+
+  context 'when building a new user without an email' do
+    let(:user) {build(:user, email: nil)}
+    it 'is invalid' do
+      user.valid?
+      _(user.errors[:email]).must_include I18n.t('activerecord.errors.messages.blank')
+    end
+  end
+
+  context 'when building a new user with panda in email' do
+    let(:user) {build(:user, email: "#{panda_panda}@panda.org")}
+    it 'is invalid' do
+      user.valid?
+      _(user.errors[:email]).must_include I18n.t('activerecord.errors.messages.invalid')
+    end
+  end
+
+  context 'when building a new user with an invalid email' do
+    let(:user) {build(:user, email: "foo@bar@com")}
+    it 'is invalid' do
+      user.valid?
+      _(user.errors[:email].length).must_equal 1
+    end
+  end
+
+  context 'when building a new user with a valid email' do
+    let(:user) {build(:user, email: "valid@example.net")}
+    it 'is valid' do
+      user.valid?
+      _(user.errors[:email].length).must_equal 0
+    end
+  end
+
+  context 'email format validation on create' do
+    it 'allows creation with a valid email address' do
+      assert_creates(User) do
+        create(:user, email: 'valid@example.net')
+      end
+    end
+
+    it 'raises an error when email is invalid' do
+      error = _ {create(:user, email: 'invalid@incomplete')}.must_raise ActiveRecord::RecordInvalid
+      _(error.message).must_equal 'Validation failed: Email does not appear to be a valid e-mail address'
+    end
+  end
+
+  describe 'email uniqueness validation' do
+    let(:good_data) do
+      {
+        email: 'foo@bar.com',
+        password: 'foosbars',
+        name: 'tester',
+        user_type: User::TYPE_STUDENT,
+        age: 28
+      }
```

Comment:
Given that we're now using `good_data` for only email validation tests, can it be simplified?

Reply:
I leverage existing factories for these two scenarios instead. Let me know if it looks better to you.